src/App.tsx

```diff
@@ -20,17 +25,134 @@ class App extends React.Component<{}, State> {
       './img/9.png',
       './img/10.png',
     ],
+
+    step: 2,
+    frameSize: 3,
+    itemWidth: 130,
+    animationDuration: 1000,
+    infinite: false,
   };
 
   render() {
-    const { images } = this.state;
+    const {
+      images,
+      step,
+      frameSize,
+      itemWidth,
+      animationDuration,
+      infinite,
+    } = this.state;
+
+    const handleInputs = (event: React.ChangeEvent<HTMLInputElement>) => {
+      const currentState = event.currentTarget.id;
+      const stateValue = +event.currentTarget.value;
+
+      switch (currentState) {
+        case 'itemWidth':
+          this.setState({ itemWidth: +stateValue });
+          break;
+        case 'frameSize':
+          this.setState({ frameSize: +stateValue });
+          break;
+        case 'step':
+          this.setState({ step: +stateValue });
+          break;
+        case 'animationDuration':
+          this.setState({ animationDuration: +stateValue });
+          break;
+        case 'infinite':
+          this.setState((prevState) => ({
+            infinite: !prevState.infinite,
+          }));
+          break;
+        default:
+          this.setState({ animationDuration: +stateValue });
+      }
+    };
```

Comment:
You can just use like that
```suggestion
     this.setState(current => ({...current, [currentState]: +stateValue }))
   };
```